src/fidesops/service/privacy_request/request_runner_service.py

```diff
@@ -32,26 +34,33 @@ def __init__(
         self.db = db
         self.privacy_request = privacy_request
 
-    def run(self) -> None:
+    def submit(self) -> Awaitable[None]:
+        """Run this privacy request in a separate thread."""
+        return run_async(self.run, self.privacy_request.id)
+
+    def run(self, privacy_request_id: str) -> None:
         # pylint: disable=too-many-locals
         """
         Dispatch a privacy_request into the execution layer by:
-            1. Generate a graph from all the currently configured datasets
+            1. Generate a graph from all the currently , configured datasets
             2. Take the provided identity data
             3. Start the access request / erasure request execution
             4. When finished, upload the results to the configured storage destination if applicable
         """
 
-        logging.info(f"Dispatching privacy request {self.privacy_request.id}")
-        self.start_processing(db=self.db)
+        _sessionmaker = get_db_session()
+        session = _sessionmaker()
```

Comment:
I dont think we have a comprehensive way of dealing with this, but at the very least I can probably mitigate this by adding a close(). WIll look into it.

Reply:
I have not hit this error, but I can see how hanging connections can be an issue. Apart from the fix, I'm curious as to how you're managing to isolate this, since the suite pulls everything down on every run.

Reply:
I'm not doing this with integration test, I ran `make integration-env` locally, and then I used the postman collection that I've created (in the repo) to create all the resources locally necessary to make a Privacy Request (there's a folder called Minimum API Calls to create a Privacy Request).

I then just submitted a few POST /privacy-request calls back-to-back and got the connections over 100.